admin/src/Utils/NaiveDateTime.elm

```diff
@@ -182,11 +187,34 @@ minutes raw =
         Nothing
 
 
+split : DateTime -> ( Date, Time )
+split (DateTime date time) =
+    ( date, time )
+
+
 dateFromExternal : External.Date -> Date
 dateFromExternal external =
     Date { year = External.year external, month = External.month external, day = External.day external }
 
 
+fromPosix : Time.Zone -> Time.Posix -> DateTime
+fromPosix zone posix =
+    DateTime (Date { year = Time.toYear zone posix, month = Time.toMonth zone posix, day = Time.toDay zone posix })
+        (Time { hour = Time.toHour zone posix, minute = Time.toMinute zone posix })
+
+
+now : Task x DateTime
+now =
+    Task.map2
```

Comment:
why can't we do `fromPosix Time.here Time.now`, elm is confusing me xD

Reply:
`Time.now` returns a `Task x Time`, but `fromPosix` needs a `Time`. The way to extract the `Time` from a `Task x Time` is to map it with `Task.map`. Since we need to get the current time zone, too, we need to do the same for `Time.here` which returns `Task x Zone`, and we can extract both into the same context with `Task.map2`.